public/pages/utils/constants.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { SORT_DIRECTION } from '../../../server/utils/constants';
+
+export enum DETECTOR_STATE {
+  DISABLED = 'Disabled',
+  INIT = 'Initializing',
+  RUNNING = 'Running',
+  INIT_FAILURE = 'Initialization failure',
+  UNKNOWN_FAILURE = 'Unknown failure',
+}
+
+export enum DETECTOR_STATE_COLORS {
+  DISABLED = 'subdued',
+  INIT = '#0000cc',
+  RUNNING = 'success',
+  INIT_FAILURE = 'danger',
+  UNKNOWN_FAILURE = 'danger',
+}
+
+export const mapToColor = (state: string) => {
```

Comment:
nit: why not use object or Map?

Reply:
I definitely should, I'm not sure why I didn't. Will change